lib/puppet/defaults.rb

```diff
@@ -126,6 +126,19 @@ def self.default_diffargs
         munge(value)
         value.to_sym
       end
+    },
+  :disable_i18n => {
+      :default => false,
+      :type    => :boolean,
+      :desc    => "If true, turns off all translations of Puppet and module
+        log messages, which affects error, warning, and info log messages,
+        as well as any translations in the report and CLI.",
+      :hook    => proc do |value|
+        if value
```

Comment:
What would it take to make this fully togglable? Re-run initialize (as we do in puppet.rb)?

Reply:
Probably ok to leave that out for now if there's additional complexity there.

Reply:
Can you generally change settings without restarting Puppet? It's easy to add stuff to `gettext/config` to allow going backwards, but I guess I don't understand the use case for that (because I don't know a lot about the intricacies about how puppet settings work).

Reply:
Not generally. So seems fine. I think you can call initialize_settings multiple times, so it *could* happen.

Reply:
Probably can't test in specs without any changes. We should target that for the 5.3.x branch if this goes in.

Reply:
👍

Reply:
This might be evaluated multiple times if set in different sections. I'm investigating.

Reply:
That appears to be the case. If I set it to `true` in `main`, then `false` in `agent`, an agent run in a japanese locale ends up in english.

Reply:
What is the right way to handle that? That seems like that would be the case for all settings like this.

Reply:
Not the worst bug. Worth fixing if we can do it in the next hour or so, but otherwise we could leave it for a z release.

Reply:
That's just how settings work (right now). We evaluate by section, which includes evaluating the hook.

Reply:
It looks like we might be running into issues re-loading the non-stubbed gettext methods once we have stubbed them out with `require 'puppet/gettext/stubs'`. I'll keep looking into it, but I wouldn't block merging this on that.

Reply:
Ok, this may be a non-issue. Our testing assumed `puppet agent -t` and `puppet apply` used the `agent` section, when they actually use the `user` section.

Reply:
+1 to not getting carried away fixing things related to the settings subystem implementation (which is hairy and has some interesting behaviors)